src/Compression/CompressionCodecZSTDQAT.cpp

```diff
@@ -0,0 +1,114 @@
+#ifdef ENABLE_ZSTDQAT_COMPRESSION
+#include <CompressionCodecZSTD.h>
+#include <Compression/CompressionInfo.h>
+#include <Compression/CompressionFactory.h>
+#include <zstd.h>
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Common/typeid_cast.h>
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+#include <qatseqprod.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPRESS;
+    extern const int ILLEGAL_SYNTAX_FOR_CODEC_TYPE;
+    extern const int ILLEGAL_CODEC_PARAMETER;
+}
+
+class CompressionCodecZSTDQAT : public CompressionCodecZSTD
+{
+public:
+    static constexpr auto ZSTDQAT_SUPPORTED_MIN_LEVEL = 1;
+    static constexpr auto ZSTDQAT_SUPPORTED_MAX_LEVEL = 12;
```

Comment:
Is maximum level = 12 a hardware restriction (i.e. unsupported) or does QAT merely become too slow with this level?

Reply:
QAT Hardware only supports compression levels L1 to L12  [limitations](https://github.com/intel/QAT-ZSTD-Plugin?tab=readme-ov-file#limitations)

Reply:
Done

Reply:
Let's put in a comment above ZSTDQAT_SUPPORTED_MIN_LEVEL/ZSTDQAT_SUPPORTED_MAX_LEVEL so future readers of the code will know right away instead of having to dig the code history and end up reading our conversation here.

Reply:
Done